### PR TITLE
docs: fix headless check

### PR DESCRIPTION
**Pull request by mweibel**
fixes godotengine/godot#84019

Please note: the c# code I didn't actually check since my project is gdscript based. LMK if there's a problem with it.

### CODE REVIEW

**Comment by AThousandShips** [comment]
You should target `master` not `4.1`, changes can then be cherry picked, see [here](https://docs.godotengine.org/en/latest/contributing/documentation/contributing_to_the_documentation.html#contributing-changes).

If you need more information about the update process please see: [PR Workflow - Updating your branch](https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#updating-your-branch).

**Comment by mweibel** [comment]
@AThousandShips oh yeah - sorry was uninentional. I was wondering why that happened as the main branch is master and not 4.1: the edit link on godot's documentation page is pointing to it's 4.1 branch and not to master. Makes sense I guess to avoid potential conflicts but it might improve life of docs contributors if it would point to master instead. Just as an idea, I'm sure there's a reason behind it :).

File: tutorials/export/exporting_for_dedicated_servers.rst [hunk 1]
```diff
@@ -229,7 +229,7 @@ main scene (or an autoload)'s ``_ready()`` method:
 
     using System.Linq;
 
-    if (OS.GetCmdlineArgs().Contains("--headless"))
+    if (DisplayServer.get_name() == "--headless")
```
Comment (Calinou) [inline tutorials/export/exporting_for_dedicated_servers.rst:232]:
```suggestion
    if (DisplayServer.GetName() == "headless")
```

File: tutorials/export/exporting_for_dedicated_servers.rst [hunk 2]
```diff
@@ -217,7 +217,7 @@ main scene (or an autoload)'s ``_ready()`` method:
 .. tabs::
  .. code-tab:: gdscript
 
-    if "--headless" in OS.get_cmdline_args():
+    if DisplayServer.get_name() == "headless:
```
Comment (Calinou) [inline tutorials/export/exporting_for_dedicated_servers.rst:220]:
```suggestion
    if DisplayServer.get_name() == "headless":
```

**Comment by mweibel** [comment]
thanks @Calinou - and sorry for the typos :( I was a little tired it seems.

**Review by Calinou — APPROVED** [review]
Thanks!

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.1.

**Pull request merged**